static/js/tools/timeline/util.ts

```diff
@@ -42,13 +47,14 @@ export function getTokensFromUrl(name: string, sep: string): Set<string> {
   return tokens;
 }
 
-export function setTokensToUrl(
-  name: string,
-  sep: string,
-  tokens: Set<string>
-): void {
+export function setTokensToUrl(tokensMap: Record<string, TokenSet>): void {
```

Comment:
maybe leave the function as it was, so all the functions in this file have consistent signatures. I think the client can just do a for loop to accomplish the change here?

Reply:
I made this change so that the window.location.hash is only set once even if different params need to be updated. If the client does this in a for loop, then the window.location.hash will be set each time, which will cause an intermediate state in the window history..

Reply:
I see you want to set the url once. In that case can make the input a list of this struct:

export interface TokenInfo {
  name: string;
  sep: string;
  tokens: Set<string>;
}

Reply:
done